numpy/linalg/_linalg.py

```diff
@@ -3174,6 +3174,128 @@ def cross(x1, x2, /, *, axis=-1):
     return _core_cross(x1, x2, axis=axis)
 
 
+# cross2d
+
+def _cross2d_dispatcher(x1, x2, /, *, axis=None):
+    return (x1, x2,)
+
+
+@array_function_dispatch(_cross2d_dispatcher)
+def cross2d(x1, x2, /, *, axis=-1):
+    """
+    Returns the cross product of two (arrays of) vectors of dimension 2.
+
+    This is a dimension 2 version of `linalg.cross`.
+    As the dimension of both `x1` and `x2` is 2, the third
+    component of each input vector is assumed to be zero.
+    The z-component of the cross product is calculated
+    accordingly and returned as a scalar.
+
+    If ``x1`` and/or ``x2`` are multi-dimensional arrays, then
+    the scalar cross-product of each pair of corresponding 2-element vectors
+    is independently computed.
+
+    This function is Array API compatible.
+
+    Parameters
+    ----------
+    x1 : array_like
+        The first input array.
+    x2 : array_like
+        The second input array. Must be compatible with ``x1`` for all
+        non-compute axes. The size of the axis over which to compute
+        the scalar cross-product must be the same size as the respective axis
+        in ``x1``.
+    axis : int, optional
+        The axis (dimension) of ``x1`` and ``x2`` containing the vectors for
+        which to compute the scalar cross-product. Default: ``-1``.
+
+    Returns
+    -------
+    out : ndarray
+        An array containing the scalar cross products.
+
+    Raises
+    ------
+    ValueError
+        When the dimension of the vector(s) in `x1` and/or `x2` does not
+        equal 2.
+
+    See Also
+    --------
+    cross : Vector cross product of 3 dimensional vectors
+    inner : Inner product
+    outer : Outer product
+    linalg.det: A similar floating point option
+
+    Examples
+    --------
+    Scalar cross product of 2-dimensional vectors:
+
+    >>> x = [1, 2]
+    >>> y = [4, 5]
+    >>> np.linalg.cross2d(x, y)
+    array(-3)
+
+    Multiple scalar cross products. Note the direction of the cross
+    product is defined by the *right-hand rule*, and
+    the cross product of a vector with itself is zero.
+
+    >>> x = np.array([[1, 2], [3, 4], [5, 6]])
+    >>> y = np.array([[3, 4], [1, 2], [5, 6]])
+    >>> np.linalg.cross2d(x, y)
+    array([-2,  2,  0])
+
+    The choice of axis matters:
+
+    >>> u = [[1, 2], [3, 4]]
+    >>> v = [[5, 6], [7, 8]]
+    >>> np.linalg.cross2d(u, v)
+    array([-4, -4])
+    >>> np.linalg.cross2d(u, v, axis=0)
+    array([-8, -8])
+
+    Compare `~linalg.cross2d` with `~linalg.det`:
+
+    >>> u = [[1, 2], [3, 4], [-1, 5]]
+    >>> v = [[5, 6], [7, 8], [-2, 9]]
+    >>> np.linalg.cross2d(u, v)
+    array([-4, -4,  1])
+    >>> stacked_array = np.stack((u, v), axis=1)
+    >>> np.linalg.det(stacked_array)
+    array([-4., -4.,  1.])
+
+    """
+    x1 = asanyarray(x1)
+    x2 = asanyarray(x2)
+
+    axis1, axis2 = (axis,) * 2
```

Comment:
Would
```suggestion
    axis1 = axis2 = axis
```
work here?  I'm not sure if that would be in line with the style guide or not

Reply:
It would work, or just two lines. I got the code I used by just copying what was in cross (3d). Not sure what the better `style` option is, and would happily let a developer change it as they see fit.